src/Sulu/Bundle/ContentBundle/Resources/config/search.xml

```diff
@@ -35,12 +35,9 @@
 
         <!-- Rebuild structure index on massive:search:index:rebuild -->
         <service id="sulu_content.search.event_listener.search" class="%sulu_content.search.event_listener.reindex.class%">
-            <argument type="service" id="sulu.phpcr.session" />
-            <argument type="service" id="sulu.content.mapper" />
+            <argument type="service" id="sulu_document_manager.document_manager" />
+            <argument type="service" id="sulu_document_manager.document_inspector" />
```

Comment:
Well, one of the options we have is having a DocumentInspector instance for each type of document (e.g. `getWebspace()` does not apply to snippets). In that case it may be beneficial to have a factory `getDocumentInspector($document)` which returns either a base inspector or  a specific one if it exists. In that case the DocumentManager might be the place for such a factory.